### PR TITLE
[14.0][MIG] sale_order_type

### PR DESCRIPTION
**Pull request by newtratip**
- Change `with_context(force_company=record.company_id.id)` to `with_company(record.company_id)`
- Change account.move (type -> move_type)
- Check account.move (function : _compute_sale_type_id) If create invoice from sale order, sale type will not computed
- Send sale type to invoice if create invoice with down payment

cc: @ecosoft-odoo

### CODE REVIEW

File: sale_order_type/models/account_move.py [hunk 1]
```diff
@@ -12,28 +12,32 @@ class AccountMove(models.Model):
...
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        ondelete="restrict",
         copy=True,
     )
 
     @api.depends("partner_id", "company_id")
     def _compute_sale_type_id(self):
+        # If create invoice from sale order, sale type will not computed.
+        if not self._context.get("default_move_type", False) or self._context.get(
```
Comment (rousseldenis) [inline sale_order_type/models/account_move.py:24]: Prefer using self.env.context following https://github.com/odoo/odoo/blob/14.0/odoo/models.py#L4972

File: sale_order_type/__init__.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 from . import models
+from . import wizard
```
Comment (rousseldenis) [inline sale_order_type/__init__.py:4]: Following OCA conventions, use wizards instead https://github.com/OCA/maintainer-tools/tree/master/template/module

**Review by rousseldenis — CHANGES_REQUESTED** [review]
LGTM. Could you attend comments ? I would have added a test for the non covered part too.

**Comment by andreampiovesana** [comment]
or record.partner_id.commercial_partner_id.with_context(
                        force_company=record.company_id.id
                    ).sale_type
Context key 'force_company' is no longer supported. Use with_company(company) instead.

**Review by danimv5 — APPROVED** [review]
Functional review. LGTM!!

**Comment by pedrobaeza** [comment]
This should include #1531

**Comment by brendapaniagua** [comment]
Functional review. LGTM!

**Review by chienandalu — CHANGES_REQUESTED** [review]
Include https://github.com/OCA/sale-workflow/pull/1553 as well, please

**Comment by BT-cserra** [comment]
Hello everyone here! I created this other PR (https://github.com/OCA/sale-workflow/pull/1646) building on this one, that merges the requested fixes (https://github.com/OCA/sale-workflow/pull/1531 and https://github.com/OCA/sale-workflow/pull/1553). @newtratip I can do a PR over your branch if you prefer.

**Comment by newtratip** [comment]
@BT-cserra yes, you can do it.

My PR is closed.

**Comment by BT-cserra** [comment]
@newtratip my idea by doing "a PR over your branch" was that you merged my changes into yours and you updated this PR. Here it is: https://github.com/ecosoft-odoo/sale-workflow/pull/2

**Comment by rousseldenis** [comment]
> @newtratip my idea by doing "a PR over your branch" was that you merged my changes into yours and you updated this PR. Here it is: [ecosoft-odoo#2](https://github.com/ecosoft-odoo/sale-workflow/pull/2)

@BT-cserra I think your PR on ecosoft needs a rebase on his branch (too much commits in your PR)

**Comment by BT-cserra** [comment]
> @BT-cserra I think your PR on ecosoft needs a rebase on his branch (too much commits in your PR)

@rousseldenis I think that's because I rebased it from origin/14.0 also.

**Review by anajuaristi — APPROVED** [review]

**Comment by BT-cserra** [comment]
I've been given feedback [on the other PR I did](https://github.com/OCA/sale-workflow/pull/1646) and there [I've fixed a bug](https://github.com/OCA/sale-workflow/commit/d53ccc6af7a5d70910c01c28bd49d843dc58d523) that happened when using sequences that depended on dates (thanks @ioans73 for finding it). 

@newtratip @rousseldenis I don't know how to proceed here. We have 3 PRs: this one, [the one I did at first](https://github.com/OCA/sale-workflow/pull/1646) and [the one I did on @newtratip 's branch](https://github.com/ecosoft-odoo/sale-workflow/pull/2) that is lacking a rebase on 14.0 and [the fix I just did](https://github.com/OCA/sale-workflow/commit/d53ccc6af7a5d70910c01c28bd49d843dc58d523). 

I'd go with https://github.com/OCA/sale-workflow/pull/1646, because has the migration, the rebase and the fixes (including the last one found by @ioans73). But I don't mind closing it and using this one. The thing is I don't have access to this branch and we are interested in this module to be in 14.0.

How can we move this forward?

**Comment by newtratip** [comment]
@BT-cserra, Sorry for my late reply, I think my PR should be closed and should go to your PR #1646 because your PR fixed bug.

**Pull request closed without merging**